.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,23 @@
+name: CI (pre-commit)
+
+on:
+  # Triggers the workflow on push or pull request events
+  push:
+    branches: [master]
```

Comment:
Looks like technically this repo has a `main` branch and not a `master` though there may be historical redirects that may allow this yaml to work?